### ISSUE TITLE
Deconvolution3D output specification and example

### ISSUE THREAD

**Issue by caseyalan** [farizrahman4u/keras-contrib#82]
Hi,

First of all, you are all life savers -- I needed Deconvolution3D to try and make a video-GAN!

I'm having a bit of trouble understanding the doc string examples (using tensorflow backend but applies to both). Specifically, when trying to figure out the appropriate kernel/strides it is suggested to "use a dummy input and see the actual output shape." But if the specified output shape is wrong, the `.predict` won't run. So I can't use this technique to debug my layer.

This is mianly me not understanding deconvolutional math, but any suggestions on how to use the function to gain understanding would be appreciated!

For reference, I'm referring to

>      # apply a 3x3x3 transposed convolution
        #  with stride 2x2x2 and 3 output filters on a 12x12x12 image:
        model = Sequential()
        model.add(Deconvolution3D(3, 3, 3, 3, output_shape=(None, 25, 25, 25, 3),
                                  strides=(2, 2, 2),
                                  padding='valid',
                                  input_shape=(12, 12, 12, 3)))
        model.summary()
        # we can predict with the model and print the shape of the array.
        dummy_input = np.ones((32, 12, 12, 12, 3))
        preds = model.predict(dummy_input)
        print(preds.shape)  # (None, 25, 25, 25, 3)
